packages/flutter/lib/src/widgets/widget_state.dart

```diff
@@ -211,18 +211,10 @@ typedef WidgetPropertyResolver<T> = T Function(Set<WidgetState> states);
 ///      will be used to resolve the color in the given states.
 ///   3. Use [WidgetStateColor.fromMap] to assign a value using a [WidgetStateMap].
 ///
-/// If a [WidgetStateColor] is used for a property or a parameter that doesn't
-/// support resolving [WidgetStateProperty<Color>]s, then its default color
-/// value will be used for all states.
-///
-/// To define a `const` [WidgetStateColor], you'll need to extend
-/// [WidgetStateColor] and override its [resolve] method. You'll also need
-/// to provide a `defaultValue` to the super constructor, so that we can know
-/// at compile-time what its default color is.
-///
```

Comment:
I'm not sure it's necessary to remove all of this. Have we made updates that makes it easier to make a const constructor? Are there any pitfalls we should point out to not allow const constructors?

Reply:
Prior to this PR, the `.fromMap()` constructor created a temporary `WidgetStateProperty` object using the provided map, and resolved it using an empty set of states to initialize the `Color`'s value. IMO it's kinda pointless to do this work if these objects aren't meant to actually be used as the types they technically inherit from.

We can now do:

```dart
const color = WidgetStateColor.fromMap({
  WidgetState.error:    Colors.red,
  WidgetState.selected: Colors.blue,
  WidgetState.disabled: Colors.grey,
  WidgetState.any:      Colors.black,
});
```